pkg/utils/utils.go

```diff
@@ -33,12 +33,24 @@ func readPrivateKeyFromFile(rsaPrivateKeyLocation string) (string, error) {
 		return "", fmt.Errorf("missing file location for private key")
 	}
 
-	privateKey, err := os.ReadFile(rsaPrivateKeyLocation)
+	data, err := os.ReadFile(rsaPrivateKeyLocation)
 	if err != nil {
 		return "", fmt.Errorf("could not read private key file from file system: %w", err)
 	}
 
-	return string(privateKey), nil
+	// If file seems to be a service account JSON, try extracting private_key
+	if strings.HasSuffix(strings.ToLower(rsaPrivateKeyLocation), ".json") {
+		var sa struct {
+			PrivateKey string `json:"private_key"`
+		}
+
+		if err := json.Unmarshal(data, &sa); err == nil && sa.PrivateKey != "" {
+			return strings.ReplaceAll(sa.PrivateKey, "\\n", "\n"), nil
+		}
```

Comment:
```suggestion
		if err := json.Unmarshal(data, &sa); err != nil {
			return "", fmt.Errorf("failed to parse service account JSON: %w", err)
		}
		if sa.PrivateKey == "" {
			return "", fmt.Errorf("service account JSON does not contain private_key")
		}

		return strings.ReplaceAll(sa.PrivateKey, "\\n", "\n"), nil
```

I think this would be a little more explicit in terms of error handling. Wdyt?

Reply:
Makes sense, I agree. This version is indeed more explicit and clearer in terms of error handling.